src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,174 @@
+using LinearAlgebra: svd, eig, eigen
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) \coloneqq
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n)$, N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = N*(N+1)/2
+
+@doc doc"""
+
+"""
+struct LinearAffineMetric <: Metric end
+
+@doc doc"""
+
+"""
+struct LogEuclideanMetric <: Metric end
+
+distance(P::SymmetricPositiveDefinite{N},x,y) = distance(MetricManifold(P,LinearAffineMetric),x,y)
+function distance(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y)
+    s = real.( eigen( x,y ).values )
+    return any(s .<= eps() ) ? 0 : sqrt(  sum( abs.(log.(s)).^2 )  )
+end
+
+inner(P::SymmetricPositiveDefinite{N}, x, w, v) = inner(MetricManifold(P,LinearAffineMetric),x,w,v)
+function inner(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,w,v)
+	svd1 = svd(x)
+	U = svd1.U
+	S = svd1.S
+	SInv = Diagonal( 1 ./ S )
+	return tr( w * U * SInv * transpose(U) * v * U * SInv * transpose(U) )
+end
+
+norm(P::SymmetricPositiveDefinite{N},x,v) = sqrt( inner(P,x,v,v) )
+norm(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,v) = sqrt( inner(P,x,v,v) )
+
+function exp!(P::SymmetricPositiveDefinite{N},y,x,v) = exp!(MetricManifold(SymmetricPositiveDefinite,LinearAffineMetric),y,x,v)
+function exp!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, y, x, v)
+	svd1 = svd(x)
+	U = svd1.U
+	S = svd1.S
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U);
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * (t.*ξ.value) * xSqrtInv
+    eig1 = eigen(0.5*( T + transpose(T) ) ) # numerical stabilization
```

Comment:
I prefer `something/2` to `0.5*something` because 0.5 is a Float64 constant.

Reply:
I assume `T` is supposed to be symmetric but may not be numerically. Then I'd suggest to use `Symmetric(T)`, or even symmetrize the product one line above.

Reply:
see note above, I would like to have the mean of the upper and lower triangles, not throwing one away, After that I am fine with `Symetric`.